tests/phpunit/tests/sitemaps/sitemaps.php

```diff
@@ -0,0 +1,396 @@
+
+		// Register and create a private post type post.
+		register_post_type( 'private_cpt', array( 'public' => false ) );
+		self::factory()->post->create( array( 'post_type' => 'private_cpt' ) );
+
+		$entries = wp_list_pluck( $this->_get_sitemap_entries(), 'loc' );
+
+		// Clean up.
+		unregister_post_type( 'public_cpt' );
+		unregister_post_type( 'private_cpt' );
+
+		$this->assertContains( 'http://' . WP_TESTS_DOMAIN . '/?sitemap=posts&sitemap-subtype=public_cpt&paged=1', $entries, 'Public CPTs are not in the index.' );
+		$this->assertNotContains( 'http://' . WP_TESTS_DOMAIN . '/?sitemap=posts&sitemap-subtype=private_cpt&paged=1', $entries, 'Private CPTs are visible in the index.' );
+	}
+
+	/**
+	 * Tests getting a URL list for post type post.
+	 */
+	public function test_get_url_list_post() {
+		$providers = wp_get_sitemaps();
+
+		$post_list = $providers['posts']->get_url_list( 1, 'post' );
+
+		$expected = $this->_get_expected_url_list( 'post', self::$posts );
+
+		$this->assertEquals( $expected, $post_list );
+	}
+
+	/**
+	 * Tests getting a URL list for post type page.
+	 */
+	public function test_get_url_list_page() {
+		// Short circuit the show on front option.
+		add_filter( 'pre_option_show_on_front', '__return_true' );
+
+		$providers = wp_get_sitemaps();
+
+		$post_list = $providers['posts']->get_url_list( 1, 'page' );
+
+		$expected = $this->_get_expected_url_list( 'page', self::$pages );
+
+		$this->assertEquals( $expected, $post_list );
+	}
+
+	/**
+	 * Tests getting a URL list for post type page with included home page.
+	 */
+	public function test_get_url_list_page_with_home() {
+		$providers = wp_get_sitemaps();
+
+		$post_list = $providers['posts']->get_url_list( 1, 'page' );
+
+		$expected = $this->_get_expected_url_list( 'page', self::$pages );
+
+		// Add the homepage to the front of the URL list.
+		array_unshift(
+			$expected,
+			array(
+				'loc' => home_url(),
+			)
+		);
+
+		$this->assertEquals( $expected, $post_list );
+	}
+
+	/**
+	 * Tests getting a URL list for post with private post.
+	 */
+	public function test_get_url_list_private_post() {
+		wp_set_current_user( self::$editor_id );
+
+		$providers = wp_get_sitemaps();
+
+		$post_list_before = $providers['posts']->get_url_list( 1, 'post' );
+
+		$private_post_id = self::factory()->post->create( array( 'post_status' => 'private' ) );
+
+		$post_list_after = $providers['posts']->get_url_list( 1, 'post' );
+
+		$private_post = array(
+			'loc' => get_permalink( $private_post_id ),
+		);
+
+		$this->assertNotContains( $private_post, $post_list_after );
+		$this->assertEqualSets( $post_list_before, $post_list_after );
+	}
+
+	/**
+	 * Tests getting a URL list for a custom post type.
+	 */
+	public function test_get_url_list_cpt() {
+		$post_type = 'custom_type';
+
+		// Registered post types are private unless explicitly set to public.
+		register_post_type( $post_type, array( 'public' => true ) );
```

Comment:
Add new assertion with `[ 'public' => true, 'rewrite' => false ]`? (Also for tax?)

Reply:
Whether a post type supports rewrites or not is pretty irrelevant for sitemap generation. It does not and should not affect whether a post type is included in the sitemap or not.

The only thing that changes is the post permalink format in the sitemap.

So instead of `<loc>https://wordpress-develop.local/?p=123</loc>` you'll get `<loc>https://wordpress-develop.local/hello-world/</loc>`